tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,283 @@
+            elif operation == "delete":
+                extension_case.extension.delete(timeout=15 * 60)
+                instance_view_extensions = self._context.vm.get_instance_view().extensions
+                if instance_view_extensions is not None and any(
+                        e.name == extension_case.extension._resource_name for e in instance_view_extensions):
+                    raise Exception(
+                        "extension {0} still in instance view after attempting to delete".format(extension_case.extension))
+            log.info(f"Operation '{operation}' for {extension_case.extension.__str__()} succeeded as expected.")
+        except Exception as error:
+            fail(
+                f"Unexpected error while trying to {operation} {extension_case.extension.__str__()}. "
+                f"Extension is allowed by policy so this operation should have completed successfully.\n"
+                f"Error: {error}")
+
+    def _operation_should_fail(self, operation, extension_case):
+        log.info("")
+        log.info(f"Attempting to {operation} {extension_case.extension}, should fail fast.")
+        if operation == "enable":
+            try:
+                timeout = (6 * 60)  # Fail fast.
+                # VirtualMachineRunCommandClient (and VirtualMachineRunCommand) does not take force_update_tag as a parameter.
+                if type(extension_case.extension) == VirtualMachineRunCommandClient:
+                    extension_case.extension.enable(settings=extension_case.settings, timeout=timeout)
+                else:
+                    extension_case.extension.enable(settings=extension_case.settings, force_update=True,
+                                                    timeout=timeout)
+                fail(
+                    f"The agent should have reported an error trying to {operation} {extension_case.extension} "
+                    f"because the extension is disallowed by policy.")
+            except Exception as error:
+                expected_msg = "Extension will not be processed: failed to run extension"
+                assert_that(expected_msg in str(error)) \
+                    .described_as(
+                    f"Error message is expected to contain '{expected_msg}', but actual error message was '{error}'").is_true()
+                log.info(f"{extension_case.extension} {operation} failed as expected")
+
+        elif operation == "delete":
+            # Delete is a best effort operation and should not fail, so CRP will timeout instead of reporting the
+            # appropriate error. We swallow the timeout error, and instead, assert that the extension is still in the
+            # instance view and that the expected error is in the agent log to confirm that deletion failed.
+            delete_start_time = self._ssh_client.run_command("date '+%Y-%m-%d %T'").rstrip()
+            try:
+                timeout = (16 * 60) # We want to wait for CRP timeout, which is 15 minutes.
```

Comment:
Marked as TODO

I initially implemented it this way, but ran into issues - subsequent operations would fail, because CRP was still waiting on the delete operation. I wasn't sure how to force CRP to quit, I will take a look at it in the next PR.